### ISSUE TITLE
Database connection not defined in debug mode

### ISSUE THREAD

**Issue by fritzmg** [contao/contao#6101]
### Affected version(s)

4.13.x

### Description

The debug mode will not work when using the Install Tool to define the database connection.

### Reproduction

1. Create a fresh Contao installation with `composer create-project contao/managed-edition . 4.13`.
2. Go to `/contao/install`, define a password and then set up the database connection.
3. Finish the installation and go into the back end.
4. Click on "Debug mode" in the top right.

The following error will occur:

<details>
  <summary>Stack trace</summary>

  ```
  Doctrine\DBAL\Exception\ConnectionException:
  An exception occurred while executing a query: SQLSTATE[3D000]: Invalid catalog name: 1046 No database selected

    at vendor\doctrine\dbal\src\Driver\API\MySQL\ExceptionConverter.php:101
    at Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert(object(Exception), object(Query))
      (vendor\doctrine\dbal\src\Connection.php:1935)
    at Doctrine\DBAL\Connection->handleDriverException(object(Exception), object(Query))
      (vendor\doctrine\dbal\src\Connection.php:1878)
    at Doctrine\DBAL\Connection->convertExceptionDuringQuery(object(Exception), 'SELECT * FROM tl_user WHERE `username`=? LIMIT 0,1', array('test'), array())
      (vendor\doctrine\dbal\src\Connection.php:1103)
    at Doctrine\DBAL\Connection->executeQuery('SELECT * FROM tl_user WHERE `username`=? LIMIT 0,1', array('test'), array())
      (vendor\contao\core-bundle\src\Resources\contao\library\Contao\Database\Statement.php:284)
    at Contao\Database\Statement->query('', array('test'))
      (vendor\contao\core-bundle\src\Resources\contao\library\Contao\Database\Statement.php:240)
    at Contao\Database\Statement->execute('test')
      (vendor\contao\core-bundle\src\Resources\contao\library\Contao\User.php:329)
    at Contao\User->findBy('username', 'test')
      (vendor\contao\core-bundle\src\Resources\contao\library\Contao\User.php:479)
    at Contao\User::loadUserByIdentifier('test')
      (vendor\contao\core-bundle\src\Framework\Adapter.php:46)
    at Contao\CoreBundle\Framework\Adapter->__call('loadUserByIdentifier', array('test'))
      (vendor\contao\core-bundle\src\Security\User\ContaoUserProvider.php:72)
    at Contao\CoreBundle\Security\User\ContaoUserProvider->loadUserByIdentifier('test')
      (vendor\contao\core-bundle\src\Security\User\ContaoUserProvider.php:87)
    at Contao\CoreBundle\Security\User\ContaoUserProvider->refreshUser(object(BackendUser))
      (vendor\symfony\security-http\Firewall\ContextListener.php:236)
    at Symfony\Component\Security\Http\Firewall\ContextListener->refreshUser(object(UsernamePasswordToken))
      (vendor\symfony\security-http\Firewall\ContextListener.php:137)
    at Symfony\Component\Security\Http\Firewall\ContextListener->authenticate(object(RequestEvent))
      (vendor\symfony\security-bundle\Debug\WrappedLazyListener.php:49)
    at Symfony\Bundle\SecurityBundle\Debug\WrappedLazyListener->authenticate(object(RequestEvent))
      (vendor\symfony\security-http\Firewall\AbstractListener.php:26)
    at Symfony\Component\Security\Http\Firewall\AbstractListener->__invoke(object(RequestEvent))
      (vendor\symfony\security-bundle\Debug\TraceableFirewallListener.php:73)
    at Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener->callListeners(object(RequestEvent), object(Generator))
      (vendor\symfony\security-http\Firewall.php:92)
    at Symfony\Component\Security\Http\Firewall->onKernelRequest(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
      (vendor\symfony\event-dispatcher\Debug\WrappedListener.php:118)
    at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
      (vendor\symfony\event-dispatcher\EventDispatcher.php:230)
    at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
      (vendor\symfony\event-dispatcher\EventDispatcher.php:59)
    at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
      (vendor\symfony\event-dispatcher\Debug\TraceableEventDispatcher.php:154)
    at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
      (vendor\symfony\http-kernel\HttpKernel.php:139)
    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
      (vendor\symfony\http-kernel\HttpKernel.php:75)
    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
      (vendor\symfony\http-kernel\Kernel.php:202)
    at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
      (public\index.php:44)
  ```
</details>

With an existing Contao installation you can reproduce it the following way:

1. Delete the `parameters.yml`.
2. Disable `APP_ENV=dev` if applicable.
3. Disable `DATABASE_URL=…` if applicable.
4. Execute `rm -r var` (i.e. make sure `var/cache/dev` and `var/jwt` is fully deleted).
5. Execute `vendor/bin/contao-setup`.
6. Go to `/contao/install` and set up the database again.
7. Go into the back end.
8. Click on "Debug mode" in the top right.

**Comment by fritzmg** [comment]
My guess is that the root cause of this issue is that the Install Tool only builds the `prod` cache, while the `dev` mode does not yet know about the `config/parameters.yml` file and thus also does not refresh itself automatically.

**Comment by aschempp** [comment]
> does not yet know about the `config/parameters.yml` file and thus also does not refresh itself automatically.

this should be fixed by https://github.com/contao/contao/pull/6109

**Comment by fritzmg** [comment]
> this should be fixed by #6109

Confirmed 👍 

Closing in favor of #6109.
